src/libmeasurement_kit/nettests/runnable.cpp

```diff
@@ -79,7 +79,11 @@ void Runnable::run_next_measurement(size_t thread_id, Callback<Error> cb,
         report::Entry entry;
         entry["test_keys"] = *test_keys;
         entry["test_keys"]["client_resolver"] = resolver_ip;
-        entry["input"] = next_input;
+        if (!(*test_keys)["input"].is_null()) {
+            entry["input"] = (*test_keys)["input"];
+        } else {
+            entry["input"] = next_input;
+        }
```

Comment:
> Am I missing something?

No, what you say is correct. I guess my original comment was incorrect in that I said `BaseClass`, which was a bit generic/misleading, and it would have been more correct to say that I was advocating for implementing this feature in `Runnable.`